File: java/ArqLibraryClassic/ArqLibraryLayered/src/bnsit/ms/arq/library/MemoryBorrowingDao.java
package bnsit.ms.arq.library;

import java.util.ArrayList;
import java.util.List;

public class MemoryBorrowingDao implements BorrowingDao {
    private static List<Borrowing> borrowings = new ArrayList<>();

    public void clear() {
        borrowings.clear();
    }

    public void init() {
        // no borrowings at start yet
    }

    public Borrowing FindByBookId(long id)
    {
        for (Borrowing borrowing : borrowings)
        {
            if (borrowing.getBookId() == id)
            {
                return borrowing;
            }
        }

        return null;
    }
    @Override
    public void insert(Borrowing borrowing) {
        borrowing.setId(Generated.borrowingId());
        borrowings.add(borrowing);
    }

    @Override
    public Borrowing findByBookId(long bookId) {
        for (Borrowing borrowing : borrowings)
        {
            if (borrowing.getBookId() == bookId)
            {
                return borrowing;
            }
        }

        return null;
    }
}

File: dotnet/ArqLibrarianClassic/ArqLibrarianClassic/Library/MemoryBorrowingDao.cs
using System.Collections.Generic;

namespace ArqLibrarianClassic.Library
{
    public class MemoryBorrowingDao : BorrowingDao
    {
        private static List<Borrowing> borrowings = new List<Borrowing>();
        
        public void Insert(Borrowing borrowing)
        {
            borrowing.Id = Generated.BorrowingId();
            borrowings.Add(borrowing);
        }

        public Borrowing FindByBookId(long id)
        {
            foreach (var borrowing in borrowings)
            {
                if (borrowing.BookId == id)
                {
                    return borrowing;
                }
            }

            return null;
        }

        public void Init()
        {
            // no borrowings at start yet
        }

        public void Clear()
        {
            borrowings.Clear();
        }
    }
}